src/test/scala/com/thirdship/libunit/units/MetricTSUnitTest.scala

```diff
@@ -0,0 +1,116 @@
+package com.thirdship.libunit.units
+
+import com.thirdship.libunit.utils.Helpers._
+import com.thirdship.libunit.{AStarConvertibleTSUnitData, ScalarConversionEdge}
+import org.scalatest.{FlatSpec, Matchers}
+
+class MetricTSUnitTest extends FlatSpec with Matchers {
+
+  private val baseUnit = "s"
+
+  private val baseParseList = List("sec".i, "second".w)
+
+  var compressedParseMap = Map(
+    baseUnit.i 	-> baseParseList,
+    "min".i 	-> List("minute".w),
+    "h".i 		-> List("hr".w, "hour".w),
+    "d".i 		-> List("day".w),
+    "y".i 		-> List("yr".w, "year".w),
+    "jiffy".i 	-> List()
+  )
+
+  var edges = List(
+    new ScalarConversionEdge(baseUnit,	baseUnit,	1),
+    new ScalarConversionEdge("min", 	baseUnit,	60),
+    new ScalarConversionEdge("h",	 	"min",		60),
+    new ScalarConversionEdge("d", 		"h",		24),
+    new ScalarConversionEdge("y", 		"d",		365.2422,	1),
+    new ScalarConversionEdge(baseUnit, 	"jiffy",	3e-24,		1)
+  )
+
+  "A MetricTSUnit" should "construct properly" in {
+    """new MetricTSUnit(baseUnit,baseParseList)""" should compile
+  }
+
+  it should "augment compressedParseMap" in {
+    val lore: MetricTSUnit = new MetricTSUnit(baseUnit,baseParseList)
+    compressedParseMap ++= lore.compressedParseMap
+
+    println(compressedParseMap.keys.map(_.baseString))
+
+    println("Test Es")
+    compressedParseMap.apply("Es".i).map(_.baseString).toSet  should be (Set("Esec","Esecond","exas","exasec","exasecond"))
```

Comment:
Ok, then your test should verify plurals. Not sure if your test will be adequate for all of the interesting combinations that our users may search for data.